rust/src/lib.rs

```diff
@@ -110,25 +112,39 @@ pub extern "C" fn zcashlc_clear_last_error() {
     ffi_helpers::error_handling::clear_last_error()
 }
 
-/// Sets up the internal structure of the data database.
+/// Sets up the internal structure of the data database.  The value for `seed` may be provided as a
+/// null pointer if the caller wishes to attempt migrations without providing the wallet's seed
+/// value.
+///
+/// Returns 0 if successful, 1 if the seed must be provided in order to execute the requested
+/// migrations, or 2 otherwise.
```

Comment:
Previously this returned 1 if successful or 0 otherwise (aka `bool`). Now, the success case is more C-like (using 0 for success), but the base error case is a somewhat-arbitrary positive value, which we would likely want to change later if we wanted to add additional reporting information. If we are going to alter that to be more C-like then we should use -1 for the base error case, not 2.
```suggestion
/// Returns 0 if successful, 1 if the seed must be provided in order to execute the requested
/// migrations, or -1 otherwise.
```